### DISCUSSION TITLE
Example to use front end framework like react together with webui?

### DISCUSSION THREAD

**Discussion by felixf4xu** [webui-dev/webui#528] [Q&A] [answered]
Hi,

I would like to see an example about how to use some front-end framework together with this webui library.

Let me take react.js as an example:

How to pack a react.js project's final output (the public html/js/css files) into .exe or .dll files?
Will it be even more complicated when react.js is written in typescript?

**Comment by AlbertShown** [comment, marked as answer]
This is the C backend React example:
https://github.com/webui-dev/webui/tree/main/examples/C/react

More backend languages are coming soon...

**Reply by felixf4xu** [reply]
nice, It's great if no additional "web server" (like pyther server) is needed

**Reply by AlbertShown** [reply]
This new [official example](https://github.com/webui-dev/webui/tree/main/examples/C/react) is actually using the built-in webui's internal web server. But you can use your own web server like Vite or Python if you want.
